### PR TITLE
Clean up legacy function upload code

### PR DESCRIPTION
**Pull request by lopert**
### WHY are these changes introduced?

Part of https://github.com/Shopify/script-service/issues/7746

### WHAT is this pull request doing?

Replaces `functionProxyRequest`, a concept that used to make sense with script service, with something more aligned with how things are done today, `getFunctionUploadUrl`

Cleans up a bunch of unused code, such as everything related to `uploadFunctionExtensions`, since functions are no longer uploaded this way.

Retains many of the relevant `uploadFunctionExtensions` relating to uploading the wasm blob by moving them under a new `describe uploadWasmBlob` section in the test file.

### Measuring impact

How do we know this change was effective? Please choose one:

- [x] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes
- [ ] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

File: packages/app/src/cli/services/deploy/upload.test.ts [hunk 1]
```diff
@@ -88,67 +87,24 @@ describe('uploadFunctionExtensions', () => {
...
-        data: {
-          uploadUrlGenerate: {
-            headers: {},
-            maxSize: '200',
-            url: uploadUrl,
-            moduleId: 'module-id',
-          },
+        uploadUrlGenerate: {
+          headers: {},
+          maxSize: '200 kb',
```
Comment (dphm) [inline packages/app/src/cli/services/deploy/upload.test.ts:92]: I guess the error message used to be "less than 200" and the test just mocked it? 😆

**Review by dphm — COMMENTED** [review]
> Cleans up a bunch of unused code, such as everything related to `uploadFunctionExtensions`, since functions are no longer uploaded this way.

Could you explain how this works now?

**Comment by lopert** [comment]
> > Cleans up a bunch of unused code, such as everything related to `uploadFunctionExtensions`, since functions are no longer uploaded this way.
> 
> Could you explain how this works now?

Sure thing! Here are some notes I took during the process.
- [deploy command](https://github.com/Shopify/cli/blob/6ea14ae8553b088c9081871c28858ae82e691664/packages/app/src/cli/commands/app/deploy.ts)
- [deploy service](https://github.com/Shopify/cli/blob/bf7c402fd436e17de2652be92eec27316eebc251/packages/app/src/cli/services/deploy.ts)
	- [exported deploy function](https://github.com/Shopify/cli/blob/86c15f8f5da2ff99e9f710980dc9a2bd6d55f8c2/packages/app/src/cli/services/deploy.ts#L52)
		- [call to ExtensionInstance.bundleConfig](https://github.com/Shopify/cli/blob/86c15f8f5da2ff99e9f710980dc9a2bd6d55f8c2/packages/app/src/cli/services/deploy.ts#L103)
			- extension-instance/[bundleConfig](https://github.com/Shopify/cli/blob/86c15f8f5da2ff99e9f710980dc9a2bd6d55f8c2/packages/app/src/cli/models/extensions/extension-instance.ts#L272)/[deployConfig](https://github.com/Shopify/cli/blob/86c15f8f5da2ff99e9f710980dc9a2bd6d55f8c2/packages/app/src/cli/models/extensions/extension-instance.ts#L145)/[functionDeployConfig](https://github.com/Shopify/cli/blob/86c15f8f5da2ff99e9f710980dc9a2bd6d55f8c2/packages/app/src/cli/models/extensions/extension-instance.ts#L150C9-L150C29)
			- upload/[uploadWasmBlob](https://github.com/Shopify/cli/blob/86c15f8f5da2ff99e9f710980dc9a2bd6d55f8c2/packages/app/src/cli/services/deploy/upload.ts#L516)
				- my new function: getFunctionExtensionUploadUrlFromPartners
	- call to [uploadextensionsBundle](https://github.com/Shopify/cli/blob/86c15f8f5da2ff99e9f710980dc9a2bd6d55f8c2/packages/app/src/cli/services/deploy.ts#L106)
	- [uploadextensionbundle](https://github.com/Shopify/cli/blob/a4fe2ca6153c1d634ffd940166b2625c68592a34/packages/app/src/cli/services/deploy/upload.ts#L133) function

File: packages/cli-kit/src/public/node/api/partners.ts [hunk 2]
```diff
@@ -38,42 +38,37 @@ export async function partnersRequest<T>(query: string, token: string, variables
...
-    api_key: apiKey,
-    query,
-    variables: JSON.stringify(variables) || '{}',
-  }
-  const proxyQuery = ScriptServiceProxyQuery
-  const res: ProxyResponse = await partnersRequest(proxyQuery, token, proxyVariables)
-  const json = JSON.parse(res.scriptServiceProxy)
-  handleDeprecations(json)
-  return json as T
+export async function getFunctionUploadUrl<T>(token: string): Promise<T> {
```
Comment (dphm) [inline packages/cli-kit/src/public/node/api/partners.ts:57]: Follow-up to [this comment](https://github.com/Shopify/cli/pull/3228/files#r1450554945), I think the sig will look like this:

```suggestion
export async function getFunctionUploadUrl(token: string): Promise<FunctionUploadUrlGenerateResponse> {
```

**Review by dphm — APPROVED** [review]
Rest of the code LGTM! Assuming you 🎩 😛

File: packages/app/src/cli/services/deploy/upload.test.ts [hunk 3]
```diff
@@ -314,441 +274,26 @@ describe('uploadFunctionExtensions', () => {
...
       await writeFile(extension.outputPath, '')
-      const uploadURLResponse: UploadUrlGenerateMutationSchema = {
-        data: {
-          uploadUrlGenerate: {
-            headers: {},
-            maxSize: '200',
-            url: uploadUrl,
-            moduleId: 'module-id',
-          },
+      const uploadURLResponse: FunctionUploadUrlGenerateResponse = {
```
Comment (andrewhassan) [inline packages/app/src/cli/services/deploy/upload.test.ts:277]: I think this test is the same as `errors if the upload of the wasm errors`. I don't see a compilation request in the code anymore.

File: packages/app/src/cli/services/deploy/upload.test.ts [hunk 4]
```diff
@@ -21,9 +20,10 @@ vi.mock('@shopify/cli-kit/node/http')
 vi.mock('@shopify/cli-kit/node/session')
 vi.mock('@shopify/cli-kit/node/crypto')
 
-describe('uploadFunctionExtensions', () => {
+describe('uploadWasmBlob', () => {
```
Comment (andrewhassan) [inline packages/app/src/cli/services/deploy/upload.test.ts:23]: Should we add a test that covers the success case?

**Pull request merged**